### ISSUE TITLE
[Test] `offer-tests` needs to be implemented

### ISSUE THREAD

**Issue by PeterSenpai** [uoft-tapp/tapp#359]

**Comment by wisemandenny** [comment]
@siefkenj  I am trying to implement a test for the following: `modify position/assignment and existing offers should not change`
(`frontend/src/tests/offers-tests.js`).

I am having trouble understanding the documentation which explains the desired behavior:
~~~
// can only create when the active offer is null or with a status of withdrawn
// if there's an active offer is not withdrawn/provisional, can't change anything of that assignment. (assignment is locked == having active offer)
// if assignment gets changed (withdrawn/provisional), there's no longer an active offer.
// two routes: admin public
// admin: accept / reject / withdraw any time
// public: accept / reject on status: pending, can't do no action when it is not pending
~~~

As far as I can understand, the desired behavior is:
~~~
if (assignment.activeOffer.status !== ("withdrawn" || "provisional")) {
  reject any changes
} else {
  accept changes
}
~~~

I don't believe this is currently implemented in the API, because I have been able to change the assignment despite there already being a pending offer for a given assignment. To fix this issue, where should I look in the code? `frontend/src/api/actions/assignments.js`? Or somewhere in the backend? If I need to work on the backend that may be an issue due to my extremely limited Ruby knowledge.

**Comment by siefkenj** [comment]
Your assessment is correct. The backend should reject such a request and the frontend should not allow such a request to be made in the first place.

See a detailed diagram of offer workflow here: https://github.com/uoft-tapp/tapp/wiki/Workflow-Diagram-for-offers

**Comment by siefkenj** [comment]
Resolved by #421
